### PR TITLE
workflow: don't set `core` field if record is auto-approved

### LINKED ISSUE #3125: workflows: core should only be set to false when a human has chosen "accept"
## Expected Behavior

When a human pushes the "Accept" button in the holding pen as opposed to "Core", the record is flagged as non-core, that is `core: false` is set in the record. On the other hand, if it's decided that the record is accepted as non-core automatically based on the arXiv category it belongs to, the record should have no `core` field in the record at all.

## Current Behavior

`core: false` is written in the record in both cases.

## Context
@fschwenn reported this by email. This translates on legacy to records having `980:NONCORE` where they shouldn’t.

### PR DESCRIPTION
**Pull request by ammirate**
This closes #3125.

Signed-off-by: Antonio Cesarano <cesarano2607@gmail.com>

## Checklist:

- [x] I have all the information that I need (if not, move to `RFC` and look for it).
- [x] I linked the related issue(s) in the corresponding commit logs.
- [x] I wrote [good commit log messages](https://github.com/torvalds/subsurface-for-dirk/blob/5f15ad5a86ada3c5e574041a5f9d85235322dabb/README#L92-L119).
- [x] My code follows the code style of this project.
- [x] I've added any new docs if API/utils methods were added.
- [x] I have updated the existing documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: inspirehep/modules/workflows/tasks/matching.py [hunk 1]
```diff
@@ -113,45 +107,35 @@ def article_exists(obj, eng):
...
+    ) > 0
 
 
 @with_debug_logging
-def set_coreness_in_extra_data(obj, eng):
-    obj.extra_data['core'] = get_coreness(obj.data).value
+def set_core_in_extra_data(obj, eng):
+    """Set `core=True` in `obj.extra_data` if the record is core"""
+    def _is_core(record):
+        return len(
```
Comment (michamos) [inline inspirehep/modules/workflows/tasks/matching.py]: the `len` thing is not needed, as empty sets are falsy and non-empty sets are truthy.

File: inspirehep/modules/workflows/tasks/matching.py [hunk 2]
```diff
@@ -113,45 +107,60 @@ def article_exists(obj, eng):
...
+        record_categories &
+        set(
+            harvested_categories.get('core') +
+            harvested_categories.get('non-core')
+        )
+    ) > 0
+
+
+def physics_data_an_is_primary_category(record):
+    record_categories = set(get_arxiv_categories(record))
```
Comment (michamos) [inline inspirehep/modules/workflows/tasks/matching.py]: if you don't cast to a `set` here, you can reuse the variable later.

Comment (ammirate) [reply inspirehep/modules/workflows/tasks/matching.py]: copy/paste leftover :+1:

File: tests/unit/workflows/test_workflows_tasks_matching.py [hunk 3]
```diff
@@ -131,12 +125,104 @@ def test_belongs_to_relevant_category_returns_true_and_is_non_core(app):
                 },
             ],
         }
+        assert validate(record['arxiv_eprints'], subschema) is None
+        assert has_fully_harvested_category(record)
+
+
+def test_has_fully_harvested_category_is_falsa_with_others_categories(app):
```
Comment (michamos) [inline tests/unit/workflows/test_workflows_tasks_matching.py]: `falsa` -> `false` :smile:

**Pull request merged**